File: stack/boj_6198.cpp
// Bad Hair Day
#include <bits/stdc++.h>
using namespace std;

int n, tmp;
long long ans;
stack<int> s;

int main(void) {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> tmp;
        while (!s.empty() && tmp >= s.top())
            s.pop();
        ans += s.size();
        s.push(tmp);
    }
    cout << ans;

    return 0;
}

File: stack/boj_6198.py
'''
2021.08.25. Bad Hair Day (옥상 정원 꾸미기)
https://www.acmicpc.net/problem/6198
'''
import sys

sys.stdin = open("../input.txt", "r")

buildings, s = [], []
ans = 0
n = int(sys.stdin.readline())
for _ in range(n):
    buildings.append(int(sys.stdin.readline().rstrip()))

for i in range(n):
    while s and s[-1] <= buildings[i]:
        s.pop()         # 못 보는 건물은 pop
    ans += len(s)       # 해당 건물을 볼 수 있는 건물의 개수
    s.append(buildings[i])
print(ans)